src/ostorlab/apis/assets.py

```diff
@@ -0,0 +1,189 @@
+        variables['asset'].update(asset_type_variables)
+
+        if asset_type in ('AndroidAab', 'AndroidApk', 'File', 'IOSIpa'):
+            map_variables = _get_map_variables(asset_type)
+            data = {
+                'operations': json.dumps({'query': self.query,
+                                          'variables': json.dumps(variables)}),
+                'map': json.dumps(map_variables)
+            }
+        else:
+            data = {
+                'query': self.query,
+                'variables': json.dumps(variables)
+            }
+        return data
+
+    @property
+    def files(self) -> Optional[Dict] :
+        """Sets the file for multipart upload of the API request.
+
+        Returns:
+            The file mapping of the create asset request.
+        """
+        asset_type = type(self._asset).__name__
+        if asset_type in ('AndroidAab', 'AndroidApk', 'File', 'IOSIpa'):
+            return {'0': self._asset.content}
+        else:
+            return None
+
+    def __get_asset_variables(self, asset_type: str) -> Dict[str, Dict]:
+        """Creates asset variables for the API request, depending on the type of the asset."""
+        asset_type_variables = {}
+        if asset_type in ('AndroidAab', 'AndroidApk'):
+            asset_type_variables = {
+                'androidFile': {
+                    'file': self._asset.content
+                }
+            }
+        elif asset_type == 'IOSIpa':
+            asset_type_variables = {
+                'iosFile':{
+                    'file': self._asset.content
+                }
+            }
+        elif asset_type == 'File':
+            asset_type_variables = {
+                'file':{
+                    'content': self._asset.content,
+                    'path': self._asset.path
+                }
+            }
+        elif asset_type == 'DomainName':
+            asset_type_variables = {
+                'domain':{
+                    'domain': self._asset.name
+                }
+            }
+        elif asset_type == 'IP':
+            asset_type_variables = {
+                'ip':{
+                    'host': self._asset.host,
+                    'mask': self._asset.mask,
+                    'version': self._asset.version
+                }
+            }
+        elif asset_type == 'IPv4':
+            asset_type_variables = {
+                'ipV4':{
+                    'host': self._asset.host,
+                    'mask': self._asset.mask,
+                    'version': self._asset.version
+                }
+            }
+        elif asset_type == 'IPv6':
+            asset_type_variables = {
+                'ipV6':{
+                    'host': self._asset.host,
+                    'mask': self._asset.mask,
+                    'version': self._asset.version
+                }
+            }
+        else:
+            raise NotImplementedError('Unknown asset type')
```

Comment:
Please make the asset type in the exception message to easily know which one in case of an exception.